crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -128,6 +152,164 @@ impl types::PaymentsAuthorizeRouterData {
+        };
+        Some(locker_response.0.payment_method_id)
+    } else {
+        None
+    };
+    Ok(pm_id)
+}
+
+pub async fn save_in_locker(
+    state: &AppState,
+    merchant_account: &storage::MerchantAccount,
+    payment_method_request: api::PaymentMethodCreate,
+) -> RouterResult<(api_models::payment_methods::PaymentMethodResponse, bool)> {
+    payment_method_request.validate()?;
+    let merchant_id = &merchant_account.merchant_id;
+    let customer_id = payment_method_request
+        .customer_id
+        .clone()
+        .get_required_value("customer_id")?;
+    match payment_method_request.card.clone() {
+        Some(card) => payment_methods::cards::add_card_to_locker(
+            state,
+            payment_method_request,
+            card,
+            customer_id,
+            merchant_account,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Add Card Failed"),
+        None => {
+            let pm_id = common_utils::generate_id(crate::consts::ID_LENGTH, "pm");
+            let payment_method_response = api::PaymentMethodResponse {
+                merchant_id: merchant_id.to_string(),
+                customer_id: Some(customer_id),
+                payment_method_id: pm_id,
+                payment_method: payment_method_request.payment_method,
+                payment_method_type: payment_method_request.payment_method_type,
+                card: None,
+                metadata: None,
+                created: Some(common_utils::date_time::now()),
+                recurring_enabled: false,           //[#219]
+                installment_payment_enabled: false, //[#219]
+                payment_experience: Some(vec![api_models::enums::PaymentExperience::RedirectToUrl]), //[#219]
+            };
+            Ok((payment_method_response, false))
+        }
+    }
+}
+
+pub fn create_payment_method_metadata(
+    metadata: Option<&pii::SecretSerdeValue>,
+    connector_token: Option<(&api::ConnectorData, String)>,
+) -> Option<serde_json::Value> {
+    connector_token.and_then(|connector_and_token| {
+        metadata.and_then(|data| {
+            let mut value_exposed = data.clone().expose();
+            value_exposed.as_object_mut().and_then(|val| {
+                val.insert(
+                    connector_and_token.0.connector_name.to_string(),
+                    serde_json::Value::String(connector_and_token.1),
+                )
+            })
+        })
+    })
+}
+
+pub fn create_new_payment_method_metadata(
+    connector_token: Option<(&api::ConnectorData, String)>,
+) -> Option<serde_json::Value> {
+    connector_token.map(|connector_and_token| {
+        let mut metadata_map = serde_json::Map::new();
+        metadata_map.insert(
+            connector_and_token.0.connector_name.to_string(),
+            serde_json::Value::String(connector_and_token.1),
+        );
+        serde_json::Value::Object(metadata_map)
+    })
+}
```

Comment:
```suggestion
pub fn create_payment_method_metadata(
    metadata: Option<&pii::SecretSerdeValue>,
    connector_token: Option<(&api::ConnectorData, String)>,
) -> Option<serde_json::Value> {
	 let mut meta = match metadata {
	    None  =>  serde_json::Map::new(),
		Some(meta) => meta.clone().expose(),
	}
    connector_token.and_then(|connector_and_token| {
    	meta.insert(
    		connector_and_token.0.connector_name.to_string(),
    		serde_json::Value::String(connector_and_token.1),
        )
   })
}
```

Reply:
Similar to this, in future it will have more fields in metadata and create also takes metadata as a input. So single function itself handle this. create metadata if not passed.